s3_file_uploads/views.py

```diff
@@ -14,14 +14,19 @@ def get_queryset(self):
         return UploadedFile.objects.filter(user=self.request.user)
 
     def create(self, request, *args, **kwargs):
-        serializer_data = {**request.data, 'user': request.user.id}
-        serializer = self.get_serializer(data=serializer_data)
-        serializer.is_valid(raise_exception=True)
-        self.perform_create(serializer)
-        headers = self.get_success_headers(serializer.data)
-        uploaded_file = serializer.instance
+        acl_serialiser = AccessControlListSerializer(data=request.data)
+        acl_serialiser.is_valid(raise_exception=True)
+        file_serializer = self.get_serializer(
+            data={**request.data, 'user': request.user.id}
+        )
+        file_serializer.is_valid(raise_exception=True)
+        self.perform_create(file_serializer)
+        headers = self.get_success_headers(file_serializer.data)
+        uploaded_file = file_serializer.instance
         uploaded_file_data = UploadedFileSerializer(instance=uploaded_file).data
-        uploaded_file_data['upload_form'] = uploaded_file.get_upload_form()
+        uploaded_file_data['upload_form'] = uploaded_file.get_upload_form(
+            acl_type=acl_serialiser.validated_data
```

Comment:
should this be?

```suggestion
            acl_type=acl_serialiser.validated_data['acl']
```

Reply:
or do I misunderstand how you pass the args down?

Reply:
I'm passing the whole dictionary down as the function that requires this is a dictionary - is this not good practice

Reply:
Updated as per jakes comments